src/t_hash.c

```diff
@@ -818,23 +818,24 @@ void genericHgetallCommand(client *c, int flags) {
     robj *emptyResp = (flags & OBJ_HASH_KEY && flags & OBJ_HASH_VALUE) ? shared.emptymap[c->resp] : shared.emptyarray;
     if ((o = lookupKeyReadOrReply(c, c->argv[1], emptyResp)) == NULL || checkType(c, o, OBJ_HASH)) return;
 
+    writePreparedClient *wpc = prepareClientForFutureWrites(c);
```

Comment:
You should handle the case when `wpc == NULL`. Since in that case no writes should be done.